src/api/common_api.js

```diff
@@ -309,52 +309,113 @@ module.exports = {
+                                ],
                             },
-                            Condition: {
+                            {
                                 type: 'object',
+                                required: ['Effect'],
                                 properties: {
-                                    StringEquals: {
-                                        $ref: '#/definitions/bucket_policy_string_condition'
+                                    Sid: {
+                                        type: 'string'
+                                    },
+                                    Principal: {
+                                        $ref: '#/definitions/bucket_policy_principal'
                                     },
-                                    StringNotEquals: {
-                                        $ref: '#/definitions/bucket_policy_string_condition'
+                                    NotPrincipal: {
+                                        $ref: '#/definitions/bucket_policy_principal'
                                     },
-                                    StringEqualsIgnoreCase: {
-                                        $ref: '#/definitions/bucket_policy_string_condition'
+                                    Action: {
+                                        $ref: '#/definitions/string_or_string_array'
                                     },
-                                    StringNotEqualsIgnoreCase: {
-                                        $ref: '#/definitions/bucket_policy_string_condition'
+                                    NotAction: {
+                                        $ref: '#/definitions/string_or_string_array'
                                     },
-                                    StringLike: {
-                                        $ref: '#/definitions/bucket_policy_string_condition'
+                                    Resource: {
+                                        $ref: '#/definitions/string_or_string_array'
                                     },
-                                    StringNotLike: {
-                                        $ref: '#/definitions/bucket_policy_string_condition'
+                                    NotResource: {
+                                        $ref: '#/definitions/string_or_string_array'
                                     },
-                                    Null: {
-                                        $ref: '#/definitions/bucket_policy_null_condition'
+                                    Effect: {
+                                        enum: ['Allow', 'Deny'],
+                                        type: 'string'
+                                    },
+                                    Condition: {
+                                        type: 'object',
+                                        properties: {
+                                            StringEquals: {
```

Comment:
in condition can we get all of these options in the same condition? Maybe also add a remark about how this schema is supposed to look, and maybe also a link to AWS docs.